Sources/NIOHTTP2/MultiplexerAbstractChannel.swift

```diff
@@ -25,21 +25,42 @@ import NIO
 /// The implementation of `Equatable` & `Hashable` on this type reinforces that requirement.
 struct MultiplexerAbstractChannel {
     private var baseChannel: BaseChannel
+    private var inboundStreamStateInitializer: InboundStreamStateInitializer
+
+    private init(baseChannel: BaseChannel, inboundStreamStateInitializer: InboundStreamStateInitializer) {
```

Comment:
Hmm, this isn't quite what I wanted either. I wanted to pass the initializer to the relevant functions. This forces us to store the inbound stream state initializer on all streams, which is mostly extra work for us. Any reason not to just pass it to `configureInboundStream`?

We don't need to _store_ this function: we just need to _have_ it in the `init` to know what channel to create. That's totally sufficient.

Reply:
Gotcha.